cmd/cstorvolumeclaim/controller.go

```diff
@@ -156,13 +161,21 @@ func (c *CVCController) syncCVC(cvc *apis.CStorVolumeClaim) error {
 		// We choose to absorb the error here as the worker would requeue the
 		// resource otherwise. Instead, the next time the resource is updated
 		// the resource will be queued again.
-		runtime.HandleError(fmt.Errorf("cvc must be publish/attached to Node: %v", cvc))
+		runtime.HandleError(fmt.Errorf("cvc must be publish/attached to Node: %+v", cvc))
+		c.recorder.Event(cvc, corev1.EventTypeWarning,
+			Provisioning,
+			fmt.Sprintf(MessageCVCPublished, cvc.Name),
+		)
 		return nil
 	}
 
 	if cvc.Status.Phase == apis.CStorVolumeClaimPhasePending {
-		klog.Infof("provisioning cstor volume %+v", cvc)
-		cvc, err = c.createVolumeOperation(cvc)
+		klog.V(2).Infof("provisioning cstor volume %+v", cvc)
+		_, err = c.createVolumeOperation(cvc)
+		if err != nil {
+			//Record an event to indicate that any provisioning operation is failed.
+			c.recorder.Eventf(cvc, corev1.EventTypeWarning, Provisioning, err.Error())
+		}
```

Comment:
I have a small concern with existing code though it is not part of this PR
in `func updateCVCObj(cvc *apis.CStorVolumeClaim, cv *apis.CStorVolume)` we are making a call to etcd and didn't verify error I think it will be a problem when an update failed in etcd and event should be generated only after successfully updating in etcd. Ignore this comment if it is already done

Reply:
added the nil err check